ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -1225,6 +1231,8 @@ export default class ConfirmTransactionBase extends Component {
           onCancelAll={() => this.handleCancelAll()}
           onCancel={() => this.handleCancel()}
           onSubmit={() => this.handleSubmit()}
+          onSetApprovalForAll={() => this.handleSetApprovalForAll()}
```

Comment:
I'm not too worried about this since I'd imagine we're going to rebuild these confirm transaction components soon enough, but should we abstract this away from the specific case of `setApprovalForAll` so that we could potentially show warning modals with different content for different kinds of transactions?

Reply:
We should not consider refactoring this at this point, it should be considered holistically separate from (but including) this change. 

> maybe its okay

okay is probably a strong word to use here. It is the least bad option to just extend this pattern of overloading props etc, rather than trying to find a way to abstract out things without considering the entirety of the prop overloading problem and organization of confirmations. I do have opinions on confirmation restructuring as a whole, in that I thought we were going to try and use the template system for them -- but perhaps that is old thinking. It has been a **long** time since the first confirmations were introduced to use that system.  The idea there was that using the template system allowed for snaps authors to interweave their own UI into ours. We should probably come to consensus on the path forward there.

Reply:
Ok sounds goood!